### PR TITLE
XML docs for Permutation feature importance for Binary, Multi-class and Ranking trainers.

### PR DESCRIPTION
**Pull request by codemzs**
towards  #3204

### CODE REVIEW

File: src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs [hunk 1]
```diff
@@ -204,7 +204,7 @@ public static class PermutationFeatureImportanceExtensions
         /// <param name="catalog">The clustering catalog.</param>
         /// <param name="predictionTransformer">The model on which to evaluate feature importance.</param>
         /// <param name="data">The evaluation data set.</param>
-        /// <param name="labelColumnName">Label column name.</param>
+        /// <param name="labelColumnName">Label column name. The column data must be <see cref="System.UInt32"/>.</param>
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs]: Should this be <see cref="KeyDataViewType" />.

File: src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs [hunk 2]
```diff
@@ -48,7 +48,7 @@ public static class PermutationFeatureImportanceExtensions
         /// <param name="catalog">The regression catalog.</param>
         /// <param name="predictionTransformer">The model on which to evaluate feature importance.</param>
         /// <param name="data">The evaluation data set.</param>
-        /// <param name="labelColumnName">Label column name.</param>
+        /// <param name="labelColumnName">Label column name.  The column data must be <see cref="System.Single"/>.</param>
```
Comment (artidoro) [inline src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs]: >data must be <see cref="System.Single"/>. [](start = 73, length = 41)

nit: The column's data type must be a scalar of <see cref="System.Single"/>
Applies to other places as well.

File: src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs (hunk 1 above)
Comment (artidoro) [reply src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs]: Yes I agree

---
In reply to: [277186918](https://github.com/dotnet/machinelearning/pull/3455#discussion_r277186918) [](ancestors = 277186918)

File: src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs [hunk 3]
```diff
@@ -289,7 +289,7 @@ public static class PermutationFeatureImportanceExtensions
         /// <param name="catalog">The clustering catalog.</param>
         /// <param name="predictionTransformer">The model on which to evaluate feature importance.</param>
         /// <param name="data">The evaluation data set.</param>
-        /// <param name="labelColumnName">Label column name.</param>
+        /// <param name="labelColumnName">Label column name.  The column data must be <see cref="System.Single"/>.</param>
```
Comment (artidoro) [inline src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs]: > The column data must be <see cref="System.Single"/>. [](start = 61, length = 53)

I checked and both lightgbm ranking and fast tree should accept both single and KeyDataViewType label columns, maybe could you add that?

File: src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs (hunk 2 above)
Comment (natke) [reply src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs]: Did we say this for the other scalar types?

Comment (codemzs) [reply src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs]: No.

---
In reply to: [277187743](https://github.com/dotnet/machinelearning/pull/3455#discussion_r277187743) [](ancestors = 277187743)

Comment (natke) [reply src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs]: Yes, I think keep it consistent

File: docs/api-reference/io-columns-ranking.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ### Input and Output Columns
-The input label column data must be <xref:System.Single> and input group column
+The input label column data must be <xref:System.Single> or [key-typed](xref:Microsoft.ML.Data.KeyDataViewType). and input group column
```
Comment (Ivanidzo4ka) [inline docs/api-reference/io-columns-ranking.md]: >key-typed [](start = 61, length = 9)

I would go with just `[Key](url) type` or just `[Key](url)`

Comment (natke) [inline docs/api-reference/io-columns-ranking.md]: We use [key] in other places.

Comment (codemzs) [reply docs/api-reference/io-columns-ranking.md]: Done.

Comment (Ivanidzo4ka) [inline docs/api-reference/io-columns-ranking.md]: >input group column [](start = 117, length = 18)

rowGroupColumnName  is name of parameter in fasttree in lightgbm. Maybe Row group input column?

Comment (codemzs) [reply docs/api-reference/io-columns-ranking.md]: @natke thoughts?

Comment (natke) [reply docs/api-reference/io-columns-ranking.md]: Actually what does that last part of the sentence mean?

Comment (codemzs) [reply docs/api-reference/io-columns-ranking.md]: There are two inputs ranker needs, one is features column and second is group id.

File: docs/api-reference/io-columns-ranking.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ### Input and Output Columns
-The input label column data must be <xref:System.Single> or [key-typed](xref:Microsoft.ML.Data.KeyDataViewType) and input row group column data must be <xref:System.UInt32>. This trainer outputs the following columns:
+The input label column data must be <xref:System.Single> or [key-typed](xref:Microsoft.ML.Data.KeyDataViewType), the feature column must be a known-sized vector of <xref:System.Single> and input row group column data must be <xref:System.UInt32>. This trainer outputs the following columns:
```
Comment (natke) [inline docs/api-reference/io-columns-ranking.md]: Looking good. Nit: [key-typed] --> [key]

Comment (codemzs) [reply docs/api-reference/io-columns-ranking.md]: DOne.

File: docs/api-reference/io-columns-ranking.md [hunk 6]
```diff
@@ -1,6 +1,5 @@
 ### Input and Output Columns
-The input label column data must be <xref:System.Single> and input group column
-data must be <xref:System.UInt32>. This trainer outputs the following columns:
+The input label column data must be <xref:System.Single> or [key](xref:Microsoft.ML.Data.KeyDataViewType) type, the feature column must be a known-sized vector of <xref:System.Single> and input row group column must be <xref:System.UInt32>. This trainer outputs the following columns:
```
Comment (artidoro) [inline docs/api-reference/io-columns-ranking.md]: This line is very long and hard to review, could you break it up?
Also there seams to be a Uint32 still, could you change that to key....?

Comment (codemzs) [reply docs/api-reference/io-columns-ranking.md]: Is that true for grioup id as well?

---
In reply to: [277191534](https://github.com/dotnet/machinelearning/pull/3455#discussion_r277191534) [](ancestors = 277191534)

**Comment by shmoradims** [comment]
>         /// the original values for all other features. The evalution metric (e.g. micro-accuracy) is then calculated

typo: evaluation (same typo in all versions)

---
Refers to: src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs:191 in cf3f0ef. [](commit_id = cf3f0ef73259937935fa15f13018803f0b8a6f83, deletion_comment = False)

**Comment by shmoradims** [comment]
>         /// <code>ImmutableArray</code> of <code>MulticlassClassificationMetrics</code> objects is returned. See the sample below for an

use cref instead of <code/> (which doesn't render correctly):

<see cref="ImmutableArray"/> of <see cref="MulticlassClassificationMetrics"/>

same with all overloads

---
Refers to: src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs:198 in cf3f0ef. [](commit_id = cf3f0ef73259937935fa15f13018803f0b8a6f83, deletion_comment = False)

File: docs/api-reference/io-columns-ranking.md [hunk 7]
```diff
@@ -1,6 +1,5 @@
 ### Input and Output Columns
-The input label column data must be <xref:System.Single> and input group column
-data must be <xref:System.UInt32>. This trainer outputs the following columns:
+The input label column data must be <xref:System.Single> or [key](xref:Microsoft.ML.Data.KeyDataViewType) type, the feature column must be a known-sized vector of <xref:System.Single> and input row group column must be [key](xref:Microsoft.ML.Data.KeyDataViewType) type. This trainer outputs the following columns:
```
Comment (shmoradims) [inline docs/api-reference/io-columns-ranking.md]: Please replace input description with this:

The input label data type must be [key](xref:Microsoft.ML.Data.KeyDataViewType) type or <xref:System.Single>. The value of the label determines relevance, where higher values indicate higher relevance. If the label is a key, then the key index is the relevance value, where the 0-index is the least relevant. If the label is a <xref:System.Single>, larger values indicate higher relevance.

**Pull request merged**